### PR TITLE
LibAudio: Automatically write a FLAC seektable

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This contains a seekpoint every 2 seconds, allowing our own players to work better.

### CODE REVIEW

File: Userland/Libraries/LibAudio/FlacWriter.cpp [hunk 1]
```diff
@@ -124,6 +126,39 @@ ErrorOr<void> FlacWriter::set_metadata(Metadata const& metadata)
     return add_metadata_block(move(vorbis_block), 0);
 }
 
+void FlacWriter::sample_count_hint(size_t sample_count)
+{
+#define oom_warning "FLAC Warning: Couldn't use sample hint to reserve {} bytes padding; ignoring hint."
```
Comment (ADKaster) [inline Userland/Libraries/LibAudio/FlacWriter.cpp:131]: Can we just use a `constexpr StringView oom_warning = "FLAC Warning: Couldn't use sample hint to reserve {} bytes padding; ignoring hint."sv` here? I don't see a reason to use a #define.

**Review by ADKaster — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibAudio/FlacWriter.h [hunk 2]
```diff
@@ -83,6 +87,12 @@ class FlacWriter : public Encoder {
     ErrorOr<void> set_sample_rate(u32 sample_rate);
     ErrorOr<void> set_bits_per_sample(u16 bits_per_sample);
 
+    // The FLAC encoder by default tries to reserve some space for seek points,
+    // but that may not be enough if more than approximately four minutes of audio are stored.
```
Comment (gmta) [inline Userland/Libraries/LibAudio/FlacWriter.h:91]: How do we need to deal with situations where `sample_count_hint` is never invoked and we pass the `default_padding` size?

File: Userland/Libraries/LibAudio/FlacWriter.cpp [hunk 3]
```diff
@@ -124,6 +126,39 @@ ErrorOr<void> FlacWriter::set_metadata(Metadata const& metadata)
...
+
+    auto const samples_per_seekpoint = m_sample_rate * seekpoint_period_seconds;
+    auto const seekpoint_count = round_to<size_t>(static_cast<double>(sample_count) / samples_per_seekpoint);
+    // Will always be divisible by 4, which means we will never have stray padding that's not divisible by 4.
```
[the remark this answers is not in the document]
Comment (kleinesfilmroellchen) [reply Userland/Libraries/LibAudio/FlacWriter.cpp:135]: 24 is divisible by 4 as far as I know

[the remark this answers is not in the document]
Comment (gmta) [reply Userland/Libraries/LibAudio/FlacWriter.cpp:135]:
```
static constexpr size_t flac_seekpoint_size = (64 + 64 + 16) / 8;
```

Unless I'm not mathing right, this is `18`

File: Userland/Libraries/LibAudio/FlacWriter.h (hunk 2 above)
Comment (kleinesfilmroellchen) [reply Userland/Libraries/LibAudio/FlacWriter.h:91]: I forgot to make the code just not write superflous seek points, which is the fix here. Oops.

**Pull request merged**